### PR TITLE
e2e: upload build image cache to the github

### PR DESCRIPTION
**Pull request by YangKeao**
### What's changed and how it works?

1. Upload the build cache for every commit in the master/release branch.
2. Automatically download the cache from the github to accelerate the build.
3. `minikube image load` from the archive directly.

image

The build time can be shrunken to 1min25s

### CODE REVIEW

**Comment by YangKeao** [comment]
@STRRL PTAL

**Comment by STRRL** [comment]
I am curious about why we use `MartijnHols/actions-cache` instead of `actions/cache`?

**Comment by YangKeao** [comment]
> I am curious about why we use `MartijnHols/actions-cache` instead of `actions/cache`?

Because I don't want to update the cache for every Pull Request. With `MartijnHols/actions-cache`, we can upload the cache for every release and master branch, and every pull request will only read from the cache.

There is a WIP pr to make it possible to skip the "upload cache" step in the official cache action. When it's merged, we can migrate to the official one.

File: .github/workflows/e2e_test.yml [hunk 1]
```diff
@@ -12,28 +12,60 @@ jobs:
...
+      # we can switch to the official cache action
+      - name: Restore build cache
+        id: cache
+        uses: martijnhols/actions-cache/restore@main
+        with:
+          path: cache
+          # don't worry about the branch, the github cache
+          # is only accessible in the same branch (or a pull_request
+          # whose target is the branch)
+          key: e2e-image-build-cache-${{ runner.os }}
```
[the remark this answers is not in the document]
Comment (STRRL) [reply .github/workflows/e2e_test.yml:26]: rest LGTM.

[the remark this answers is not in the document]
Comment (YangKeao) [reply .github/workflows/e2e_test.yml:26]: No. According to the [docs](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache), the pr can only see the cache in the target branch, which is intuitive.

**Review by STRRL — APPROVED** [review]
LGTM! 🤩🤩

**Review by cwen0 — APPROVED** [review]
Cool! LGTM

**Pull request merged**